frontend/src/pages/modelServing/screens/metrics/TrustyChart.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { Stack, ToolbarContent, ToolbarItem, Tooltip } from '@patternfly/react-core';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import MetricsChart from '~/pages/modelServing/screens/metrics/MetricsChart';
+import ScheduledMetricSelect from '~/pages/modelServing/screens/metrics/ScheduledMetricSelect';
+import {
+  InferenceMetricType,
+  ModelServingMetricsContext,
+} from '~/pages/modelServing/screens/metrics/ModelServingMetricsContext';
+import { DomainCalculator, MetricsChartTypes } from '~/pages/modelServing/screens/metrics/types';
+
+type TrustyChartProps = {
+  title: string;
+  abbreviation: string;
+  metricType: InferenceMetricType.TRUSTY_AI_SPD | InferenceMetricType.TRUSTY_AI_DIR;
+  tooltip?: React.ReactElement<typeof Stack>;
+  thresholds: [number, number];
+  domain: DomainCalculator;
+  type?: MetricsChartTypes;
+};
+
+const TrustyChart: React.FC<TrustyChartProps> = ({
+  title,
+  abbreviation,
+  metricType,
+  tooltip,
+  thresholds,
+  domain,
+  type = MetricsChartTypes.AREA,
+}) => {
+  const THRESHOLD_COLOR = 'red';
+  const { data } = React.useContext(ModelServingMetricsContext);
+  const [selectedPayloadName, setSelectedPayloadName] = React.useState<string>();
+
+  const metricData = data[metricType].data;
+
+  //TODO: Fix this. This is a short term hack to add a property that will be provided by TrustyAI by release time.
+  metricData.forEach((x, i) => {
+    if (!x.metric?.registeredName) {
+      x.metric.registeredName = `Payload ${i}`;
+    }
+  });
+
+  React.useEffect(() => {
+    if (!selectedPayloadName) {
+      setSelectedPayloadName(metricData[0]?.metric?.registeredName);
+    }
+  }, [selectedPayloadName, metricData]);
+
+  const payloadOptions: string[] = metricData.map((payload) => payload.metric.registeredName);
+
+  const selectedPayload = metricData.find((x) => x.metric.registeredName === selectedPayloadName);
+
+  const metric = {
+    ...data[metricType],
+    data: selectedPayload?.values,
+  };
+
+  return (
+    <MetricsChart
+      title={`${title} (${abbreviation})`}
+      metrics={{
+        name: abbreviation,
+        metric: metric,
+      }}
+      domain={domain}
+      toolbar={
+        <ToolbarContent>
+          {tooltip && (
+            <ToolbarItem>
+              <div style={{ display: 'inline-block' }}>
```

Comment:
Not 100% sure which part you're referring to in these four lines: I assume it's the div with inline style? If so, I copied the tooltip skeleton from somewhere else in the UI and that came with it. I've just tested it and it's unnecessary here, so I'll remove it! Good catch.

Reply:
FWIW -- this is just how GitHub review works... If you comment on a line, it shows the lines before it. If you select a block of lines, it will group them. I'll try to aim my comments more specifically to talk about what line I am referring to -- but just an FYI.